resources/js/mp-history/content.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import ShowMoreLink from 'components/show-more-link';
+import LegacyMatchEvent from 'interfaces/legacy-match-event-json';
+import LegacyMatchGame from 'interfaces/legacy-match-game-json';
+import LegacyMatch from 'interfaces/legacy-match-json';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { bottomPageDistance } from 'utils/html';
+import { trans } from 'utils/lang';
+import UserJson from '../interfaces/user-json';
+import Event from './event';
+import Game from './game';
+
+interface Props {
+  currentGameId?: number;
+  events?: LegacyMatchEvent[];
+  hasNext: boolean;
+  hasPrevious: boolean;
+  isAutoloading: boolean;
+  loadingNext: boolean;
+  loadingPrevious: boolean;
+  loadNext: (state: any) => void; // TODO: type is temporary, will be specified better when `main.coffee` is converted to TS
+  loadPrevious: (state: any) => void; // TODO: type is temporary, will be specified better when `main.coffee` is converted to TS
+  match: LegacyMatch;
+  users: Partial<Record<number, UserJson>>;
+}
+
+interface Snapshot {
+  referenceFunc?: () => number;
+  referencePrev?: number;
+  scrollToLastEvent: boolean;
+}
+
+export interface TeamScores {
+  blue: number;
+  red: number;
+}
+
+export default class Content extends React.PureComponent<Props> {
+  scoresCache: Partial<Record<number, TeamScores>> = {};
+
+  componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<any>, snapshot?: Snapshot) {
+    if (snapshot?.scrollToLastEvent) {
+      $(window).stop().scrollTo(document.body.scrollHeight, 500);
+    } else if (snapshot?.referenceFunc) {
+      const referenceCurrent = snapshot.referenceFunc();
+      const documentScrollTopCurrent = window.scrollY;
+      const documentScrollTopTarget = documentScrollTopCurrent + referenceCurrent - snapshot.referencePrev!;
+      window.scrollTo(window.scrollX, documentScrollTopTarget);
+    }
+  }
+
+  getSnapshotBeforeUpdate(prevProps: Readonly<Props>, _: Readonly<any>): Snapshot {
```

Comment:
`_` kinda shadows global lodash. Maybe call it its actual name prevState just like in the other function (and use `unknown` too)

Reply:
> ...or just remove it altogether as it's not actually used

I'm very not used to just being able to do that 🤯 I figured since it was a react method it had to have a matching signature but I guess that's just not how js rolls is it